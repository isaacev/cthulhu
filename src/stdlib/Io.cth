intrinsic {
  fn php_print(Str) -> ();
  fn read_argv() -> [Str];
}

#[inline]
pub fn print(str: Str) -> () {
  php_print(str)
}

#[inline]
pub fn println(str: Str) -> () {
  php_print(str ++ "\n")
}

pub fn argv() -> [Str] {
  read_argv()
}
