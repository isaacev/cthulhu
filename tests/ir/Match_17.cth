use ::Io;

type Shapes =
  | Unit
  | Circle(Float)
  | Square { size: Float };

#[entry]
fn main() -> () {
  let shape = Unit;
  match shape {
    Unit      => Io::println("is unit"),
    Circle(_) => Io::println("is circle"),
    Square    => Io::println("is square"),
  };
}
