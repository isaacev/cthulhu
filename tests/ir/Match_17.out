
  PATTERN INCOMPATIBLE WITH TYPE  --------------------------------------------

     : TEST_DIR/ir/Match_17.cth
  14 |     Square    => Io::println("is square"),
     |     ^^^^^^

  The branch needs a pattern that matches:

    Square{ size: Float }

